app/src/features/Batches/BatchListItem.tsx

```diff
@@ -60,7 +60,7 @@ const BatchListItem = ({ batch }: BatchListItemType): JSX.Element => {
   const isBatchRunning = !batch.canceled_at && !batch.completed_at;
   const hasBatchStopped = batch.completed_at || batch.canceled_at;
 
-  const getBatchDuration = (batch: Batch) => {
+  const batchDuration = useMemo(() => {
     if (batchEnd) {
       const start = moment(batch.created_at);
```

Comment:
totally unrelated but maybe it's not a good idea tu use moment in a react project (unless we have a dependency which is already using it), it can impact bundle size quite a bit and its operations are not immutable.
see: https://momentjs.com/docs/
I've been using Luxon (https://moment.github.io/luxon/) as an alternative in another project and it's been a good experience.